packages/tupaia-web/src/features/Map/MapOverlays/DataVisualsLayer.tsx

```diff
@@ -30,26 +34,29 @@ export const DataVisualsLayer = ({
   hiddenValues: LegendProps['hiddenValues'];
 }) => {
   const navigateToEntity = useNavigateToEntity();
-  const { serieses, measureData, entities } = useActiveMapOverlayReport();
+  const { projectCode, entityCode } = useParams();
+  const { data: entity } = useEntity(projectCode, entityCode);
+  const { serieses, measureData, activeEntity, selectedOverlay } = useMapOverlayData(hiddenValues);
+  useEffect(() => {
+    gaEvent('Map Overlays', 'Change', selectedOverlay?.name);
+  }, [selectedOverlay]);
+  useEffect(() => {
+    gaEvent('Entity', 'Change', activeEntity?.name);
```

Comment:
Also maybe add a check here that `activeEntity` is set. Possibly also make the property you're watching `activeEntity?.name`